src/RemoteIo.cpp

```diff
@@ -228,7 +228,7 @@ namespace Exiv2
             if (rcount == 0) {
                 throw Error(kerErrorMessage, "Data By Range is empty. Please check the permission.");
             }
-            const byte* source = reinterpret_cast<const byte*>(data.c_str());
+            const auto source = reinterpret_cast<const byte*>(data.c_str());
```

Comment:
So the reason auto is used here is to avoid repeating the type name, which is already on the right. Original code was wrong. It should have been const byte *.

Reply:
Note that also this is different than the original. const auto evaluates to const byte* const, which is more correct I think. The source variable is not modified.